### ISSUE TITLE
SortableJS implementation (Docs)

### ISSUE THREAD

**Issue by stevebauman** [livewire/livewire#698]
I spent a ton of time trying to figure out how to implement [SortableJs](https://github.com/SortableJS/Sortable) (more time than I'd like to admit 😅) with Livewire. I finally figured it out and I just wanted to post it here in case anyone else tries to do this and needs a starting point. I'm also using Turbolinks, so this is compatible with it.

I'm also using the [spatie/eloquent-sortable](https://github.com/spatie/eloquent-sortable), which is being used when the `ExternalForce::setNewOrder()` is performed.

My app layout file:

```blade
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" class="h-100">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

        <!-- CSRF Token -->
        <meta name="csrf-token" content="{{ csrf_token() }}">

        <title>{{ config('app.name', 'Risk Management') }}</title>

        <!-- Scripts -->
        @stack('scripts')

        <!-- Styles -->
        @stack('styles')
    </head>
    <body class="h-100">
        @yield('body')

         <livewire:scripts></livewire:scripts>
    </body>
</html>
```

My component:

```php
<?php

namespace App\Http\Livewire;

use App\Report;
use App\ExternalForce;
use Livewire\Component;

class OrderForces extends Component
{
    use EmitsNotifications;

    /**
     * The current report.
     *
     * @var Report
     */
    public $report;

    /**
     * The report external forces.
     *
     * @var \Illuminate\Database\Eloquent\Collection
     */
    public $forces;

    /**
     * The sort order of the forces in a comma separated list.
     *
     * @var string
     */
    public $order;

    public function mount(Report $report, $forces = [])
    {
        $this->report = $report;
        $this->forces = $forces;
        $this->order = $forces->implode('id', ',');
    }

    public function save()
    {
        $order = explode(',', $this->order);

        ExternalForce::setNewOrder($order);

        $this->success('Re-ordered.');
    }

    public function render()
    {
        return view('livewire.order-forces');
    }
}
```

My view:

```blade
<div wire:ignore>
    <input name="ordered-forces" type="hidden" wire:model="order" value="{{ $order }}">

    <ul id="sortable-forces" class="list-group mb-4">
        @foreach($forces as $force)
            <li class="list-group-item handle">
                <input type="hidden" name="forces[]" value="{{ $force->id }}">

                <span class="mr-2">
                    <i class="fa fa-align-justify"></i>
                </span>

                <span class="text">{{ $force->title }}</span>
            </li>
        @endforeach
    </ul>

    <div class="d-flex justify-content-between">
        <button type="button" class="btn btn-light border" data-dismiss="modal">
            Cancel
        </button>

        <button type="button" wire:click="save" class="btn btn-primary">
            <i class="fa fa-save"></i> Save
        </button>
    </div>
</div>

@push('scripts')
    <script>
        document.addEventListener('turbolinks:load', () => {
            Sortable.create(document.getElementById('sortable-forces'), {
                handle: '.handle',
                animation: 150,
                onEnd: (event) => {
                    let ids = [];

                    $(event.from).find('input[name^="forces"]').each(function () {
                        ids.push(this.value);
                    });

                    let input = $('input[name="ordered-forces"]').get(0);

                    input.value = ids.join(',');

                    input.dispatchEvent(new Event('input'));
                }
            });
        });
    </script>
@endpush
```

The main takeaways here, is you have to use `wire:ignore` (as described in the docs) so when you drag and drop elements and the `input.dispatchEvent` call is made, Livewire doesn't re-render the component so you're list doesn't reset back to it's original state.

Basically, when rendering the component the ordered `ExternalForce` ID's are imploded into a comma separated array. I then re-build this array and then send it back to the component via the `wire:model` on the `ordered-forces` hidden input element.

Notice how I don't `wire:model` the hidden input inside of each draggable `<li>`, as it is not needed here and can't be used with Livewire.

Hope this helps someone in the future!

**Comment by bilogic** [comment]
@stevebauman any tips on how to get going with dropzone.js https://github.com/enyo/dropzone? Thank you.
